### DISCUSSION TITLE
Get PNG outer contour (outline) to clockwise simplified polygon?

### DISCUSSION THREAD

**Discussion by gcmartijn** [ImageMagick/ImageMagick#2672] [Help]
Is it possible to get the outer contour line using a transparent PNG to create a simplified clockwise points polygon?

The simplified polygon (points) can I use for collision, mouse overs, and other things inside a other program.
I don't want a convex hull.

I did combine several things I already found together but that din't work well.

This is one of the original x images (all different shapes).
![WFVso](https://user-images.githubusercontent.com/6859464/95015429-2355b500-064d-11eb-94b8-826940a72fc4.png)

![bord_converted](https://user-images.githubusercontent.com/6859464/95015377-d2de5780-064c-11eb-8c6d-05f4e1ab07fd.png)

![bordout](https://user-images.githubusercontent.com/6859464/95015379-d40f8480-064c-11eb-9e88-7e1b4815e371.png)

```
#convert "$IMAGE" -channel A -morphology EdgeOut Diamond +channel -fx 'a' -negate txt:- | tail -n +2 | sed -n 's/^\(.*\):.*none$/\1/p'

#convert "$IMAGE" -channel A -morphology EdgeOut Diamond +channel -fx 'a' -negate temp/bordout.jpg


#convert "$IMAGE" -matte -bordercolor none -border 1 -alpha extract -edge 1 -threshold 50% -depth 8 temp/bordout.png
#convert "$IMAGE" -matte -bordercolor none -border 1 -alpha extract -edge 1 -threshold 50% -depth 8 txt: | awk -F: '/white/{print $1}'


# hull
# https://imagemagick.org/script/convex-hull.php

#convert project/photoshop/production/scene/1/scene-assets/bord.png -channel A -morphology EdgeOut Diamond +channel -fx 'a' -negate txt:-
```

The question in more detail is here: https://stackoverflow.com/questions/64085745/get-the-contour-outline-from-a-png-image-with-the-correct-edges

**Comment by snibgo** [comment]
Please show what you want as an output.

If your input polygon is always comprised of rectangles, then "-morphology" can easily find the corners. But you want the corners listed in order around the polygon. Is that correct?

The example you show on stackoverflow doesn't trace the table shape. Instead, it traces the individual rectangles around the table shape, so you get multiple polygons. Did you do that deliberately?

**Reply by gcmartijn** [reply]
Well, what I want haha is this.

I have PNG files (asset game data) like the first coloured shape above.
In the 'game engine' I need to see collision and mouse over and other stuff.

Or for example use https://love2d.org/wiki/love.graphics.polygon

I don't want individual rectangles or polygons, only the corners that are needed to rebuild the shaping (a polygon).
In this case there are 12 different corners.

But another shape can be star, it could be anything.
